libgadget/exchange.c

```diff
@@ -82,34 +78,10 @@ int domain_exchange(int (*layoutfunc)(int p), int do_gc) {
         MPI_Type_commit(&MPI_TYPE_PLAN_ENTRY);
     }
 
-    /* flag the particles that need to be exported */
+    /*Structure for building a list of particles that will be exchanged*/
     ExchangePlan plan;
-    /*Build a list of particles that will be exchanged*/
-    plan.first = 0;
     plan.last = 0;
-    plan.ExchangeList = mymalloc2("exchangelist", sizeof(int) * PartManager->NumPart * omp_get_max_threads());
-    size_t *nexthr = ta_malloc("nexthr", size_t, omp_get_max_threads());
-    int **threx = ta_malloc("threx", int *, omp_get_max_threads());
-    gadget_setup_thread_arrays(plan.ExchangeList, threx, nexthr,PartManager->NumPart,omp_get_max_threads());
-
-    #pragma omp parallel for
-    for(i = 0; i < PartManager->NumPart; i++)
-    {
-        const int tid = omp_get_thread_num();
-        if(P[i].IsGarbage)
-            continue;
-        int target = layoutfunc(i);
-        if(target != ThisTask) {
-            threx[tid][nexthr[tid]] = i;
-            nexthr[tid]++;
-        }
-    }
-    /*Merge step for the queue.*/
-    plan.nexchange = gadget_compact_thread_arrays(plan.ExchangeList, threx, nexthr, omp_get_max_threads());
-    ta_free(threx);
-    ta_free(nexthr);
-    /*Shrink memory*/
-    plan.ExchangeList = myrealloc(plan.ExchangeList, sizeof(int) * plan.nexchange);
+    plan.nexchange = PartManager->NumPart;
```

Comment:
just any large number?

Reply:
I want there to be enough memory in the first iteration of ExchangeList, which gets allocated with nexchange. 

plan.ExchangeList = mymalloc2("exchangelist", sizeof(int) * plan.nexchange * omp_get_max_threads());